tests/test_cli.py

```diff
@@ -199,3 +199,93 @@ def test_order_search(bdb, stdin, prompt, search, exclude, keywords, rest):
     elif search == 'sreg':
         bdb.search_keywords_and_filter_by_tags.assert_called_with(
             keywords, regex=True, stag=stag, markers=markers, without=exclude, order=order)
+
+@pytest.mark.parametrize('json', [None, '', 'output.json'])
+@pytest.mark.parametrize('indices', [None, '', '1-10'])  # None = search
+@pytest.mark.parametrize('random', [None, 1, 3])
+@mock.patch('random.sample', return_value='sampled')
+@mock.patch('buku.print_rec_with_filter')
+@mock.patch('buku.write_string_to_file')
+@mock.patch('buku.format_json', return_value='formatted')
+@mock.patch('buku.print_json_safe')
+def test_random(_print_json_safe, _format_json, _write_string_to_file, _print_rec_with_filter, _sample,
+                bdb, stdin, prompt, random, indices, json):
```

Comment:
Testing `--random` handling when combined with `--print`, search and/or `--json`